invoiceregistry.go

```diff
@@ -164,20 +290,14 @@ func (i *invoiceRegistry) SettleInvoice(rHash chainhash.Hash) error {
 // notifyClients notifies all currently registered invoice notification clients
 // of a newly added/settled invoice.
 func (i *invoiceRegistry) notifyClients(invoice *channeldb.Invoice, settle bool) {
-	i.clientMtx.Lock()
-	defer i.clientMtx.Unlock()
-
-	for _, client := range i.notificationClients {
-		var eventChan chan *channeldb.Invoice
-		if settle {
-			eventChan = client.SettledInvoices
-		} else {
-			eventChan = client.NewInvoices
-		}
+	event := &invoiceEvent{
+		isSettle: settle,
+		invoice:  invoice,
+	}
 
-		go func() {
-			eventChan <- invoice
-		}()
+	select {
+	case i.invoiceEvents <- event:
```

Comment:
This may cause invoices to be delivered out of order, which means the clients may have holes in their history

Reply:
What will?

Reply:
execution of this method w/in a goroutine can permit reordering of the settled invoices. the settle ids will be assigned in-order, but we need to ensure they're also queued in that order